compiler/destroyer.nim

```diff
@@ -458,6 +458,11 @@ proc moveOrCopy(dest, ri: PNode; c: var Con): PNode =
     else:
       result = genCopy(c, dest.typ, dest, ri)
     result.add p(ri, c)
+  of nkStmtListExpr:
+    result = newNodeI(nkStmtList, ri.info)
```

Comment:
But that's dubious, it should be at least ``newNodeIT(nkStmtListExpr, ri.info, ri.typ)``

Reply:
IMO, I am  rewriting
```nim
a = (
   expr1
   expr2
   expr3)
```
into
```
expr1
expr2
a = expr3
```
Pretty much expression to statement lowering. In similar fashion I thought of lowering ifExpr to ifStmt, case expr to case stmt, block expr to block statement by replacing their last operations in branches.Mentioned operations are still always copying.

Do you have alternative idea?

Reply:
Ok, I checked that ``moveOrCopy`` is not supposed to produce an expression context, so your patch is fine.